packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2872,17 +2871,17 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// This includes the same information for this RenderObject as given by
   /// [toStringDeep], but does not recurse to any children.
   @override
-  String toStringShallow({
+  String? toStringShallow({
```

Comment:
we should make toStringShallow in foundation non-nullable (just remove the if, then return `?? toString()` in the return at the end after the assert)